packages/extension/src/components/DaFeed.vue

```diff
@@ -106,6 +122,176 @@ export default {
+
+      const currentPost = $refs.posts[focusedAtIndex].$el;
+
+      if (!currentPost.parentElement.nextSibling || insaneMode) return currentPost;
+      
+      const index = getElementIndexFromSiblings(currentPost);
+
+      return currentPost.parentElement.nextSibling.childNodes[index];
+    },
+
+    getAbovePost() {
+      const { focusedAtIndex, getElementIndexFromSiblings, $refs } = this
+
+      const currentPost = $refs.posts[focusedAtIndex].$el;
+
+      if ((currentPost.previousSibling && currentPost.previousSibling.nodeType !== Node.ELEMENT_NODE) ||
+          currentPost.previousSibling === null) return currentPost;
+
+      return currentPost.previousSibling;
+    },
+
+    getBelowPost() {
+      const { focusedAtIndex, getElementIndexFromSiblings, $refs } = this
+
+      const currentPost = $refs.posts[focusedAtIndex].$el;
+
+      if ((currentPost.nextSibling && currentPost.nextSibling.nodeType !== Node.ELEMENT_NODE) ||
+          currentPost.nextSibling === null) return currentPost;
+
+      return currentPost.nextSibling;
+    },
+
+    focusOnPost(index) {
+      const { $refs, focusedAtIndex, insaneMode, animateInsanePost, animateCard, removeHoverOnExistingPost } = this;
+
+      if (!$refs.posts[index]) return;
+
+      if (index === focusedAtIndex) return;
+
+      const animate = insaneMode ? animateInsanePost : animateCard;
+      
+      animate($refs.posts[index]);
+
+      removeHoverOnExistingPost();
+
+      this.focusedAtIndex = index;
+    },
+
+    animateInsanePost(post) {
+      if (post.ad) return post.$el.classList.add("hover");
+
+      const [insane] = post.$el.getElementsByClassName("insane--post");
+      
+      insane.getElementsByClassName("insane__link")[0].focus();
+      insane.classList.add("hover")
+    },
+
+    animateCard(post) {
+      post.$el.getElementsByClassName("card__link")[0].focus();
+      post.$el.classList.add("hover");
+    },
+
+    removeHoverOnExistingPost() {
+      const { focusedAtIndex, insaneMode, removeInsaneHover, removeCardHover, $refs } = this;
+
+      if (focusedAtIndex === null) return;
+
+      const removeHover = insaneMode ? removeInsaneHover : removeCardHover;
+
+      removeHover($refs.posts[focusedAtIndex]);
+
+      this.focusedAtIndex = null;
+    },
+
+    removeInsaneHover(post) {
+      if (post.ad) return post.$el.classList.remove("hover");
+
+      const [insane] = post.$el.getElementsByClassName("insane--post");
+      insane.classList.remove("hover");
+    },
+
+    removeCardHover(post) {
+      post.$el.classList.remove("hover");
+    },
+
+    getFirstChildElement(parent) {
+      const child = Array.from(parent.childNodes).find(child =>
+          (child.previousSibling === null && child.nodeType == Node.ELEMENT_NODE) ||
+          (child.previousSibling && child.previousSibling.nodeType !== Node.ELEMENT_NODE));
+
+      const index = this.getPostElementIndex(child);
+
+      return index >= 0 ? child : child.nextSibling;
+    },
```

Comment:
Didn't realize there's a `firstElementChild` api. Lol

Reply:
Could you please make sure to use it before I start the review? :pray: